### PR TITLE
Mutex and RwLock need RefUnwindSafe too

### PR DESCRIPTION
**Pull request by habnabit**
Incomplete, because I don't know what the appropriate stability annotation is here, but this is an attempt to bring the documentation for `std::panic` in line with reality. Right now, it says:

> Types like `&Mutex<T>`, however, are unwind safe because they implement poisoning by default.

But only `Mutex<T>`, not `&Mutex<T>`, is unwind-safe.

### CODE REVIEW

File: src/libstd/panic.rs [hunk 1]
```diff
@@ -227,6 +227,12 @@ impl<T: ?Sized> !RefUnwindSafe for UnsafeCell<T> {}
 #[stable(feature = "catch_unwind", since = "1.9.0")]
 impl<T> RefUnwindSafe for AssertUnwindSafe<T> {}
 
+// XXX: Obviously wrong, but what should it be?
+#[stable(feature = "catch_unwind", since = "1.9.0")]
```
Comment (sfackler) [inline src/libstd/panic.rs]: You can just make up a new feature name and change the since version to 1.12.0.

Comment (habnabit) [inline src/libstd/panic.rs]: Done.

**Comment by sfackler** [comment]
LGTM but ping @aturon

**Comment by alexcrichton** [comment]
@bors: r+ fbec232df1de37fce9badfe56c4a1d7c0050f3d5

Hm yes this was indeed the original intention! Not sure how it slipped through the cracks...

**Pull request merged**